src/AutoRest.CSharp/AutoRest/Plugins/CSharpProj.cs

```diff
@@ -44,6 +84,17 @@ public async Task<bool> Execute(IPluginCommunication autoRest)
 
             var context = new BuildContext(codeModel, configuration, null);
 
+            if (configuration.SkipCSProjPackageReference)
+            {
+                _csProjContent = string.Format(_csProjContent, "");
```

Comment:
let's store this into a local instead of the field.

Reply:
What is the advantage of making `_csProjContent` and `_csProjPackageReference` local variables? If we move these strings inside the method then it will create new string for each `Execute` call.

Reply:
What is the advantage of making `_csProjContent` and `_csProjPackageReference` local variables? If we move these strings inside the method then it will create new string for each method call.

Reply:
Imagine `public async Task<bool> Execute(IPluginCommunication autoRest)` was called multiple times with different values of `autoRest`. Every call after the first one won't work correctly because the method's side effects leak into the containing class.


Storing the content into the variable would make it so you can invoke this method as many times as you want.

This might not matter much in this case but controlling side effects is a good habit in type design in general.

Reply:
Yeah! I agree with you, actually I wanted to create a new string inside the method, instead of using the same `_csProjContent` and assigning values in that. Do you think this would resolve your concern If I create a new string, instead of moving the whole the properties inside?

Reply:
Yeah, that works.

I mean something like this:

```C#
  var version = GetVersion();
  var csProjPackageReference = string.Format(_csProjPackageReference, version);
  var csProjContent = string.Format(_csProjContent, csProjPackageReference);
```

Reply:
Yeah! that's what I meant.